### PR TITLE
fix LibrePhotos/librephotos #689

### LINKED ISSUE #689: email is not accepted
# 🐛 Bug Report

* [ ] 📁 I've Included a ZIP file containing my librephotos `log` files
* [X] ❌ I have looked for similar issues (including closed ones)
* [ ] 🎬 (If applicable) I've provided pictures or links to videos that clearly demonstrate the issue 

## 📝 Description of issue:
During the registering process my email format of aaa.**b**.cccc@xxx is not accepted as a valid email address.
Through trial and error I have found out that aaa.**bb**.cccc@xxx is accepted as a valid email address

Probably related to this closed issue 
https://github.com/LibrePhotos/librephotos/issues/637

### PR DESCRIPTION
**Pull request by AudreyBeard**
Fixes https://github.com/LibrePhotos/librephotos/issues/689 by adding `\.?` in the middle of the regex.

I also want to flag that this regex is generally pretty brittle and should probably get an overhaul. I'm not a frontend eng and don't have much experience with email validation in general, but I'm wondering if there's a more-robust solution than an increasingly complex regex. I'll open an issue in https://github.com/LibrePhotos/librephotos since it looks like issues are turned off on this repo

### CODE REVIEW

**Comment by AudreyBeard** [comment]
lol apparently SonarCloud agrees

**Comment by AudreyBeard** [comment]
I'm not sure why https://github.com/LibrePhotos/librephotos-frontend/pull/111 passed SonarCloud smell with an A, while this one fails with a D after such a small addition.

In any case, I don't think it's possible to make a solid regex-based email validator that is as simple as SonarCloud wants

File: src/util/util.ts [hunk 1]
```diff
@@ -5,7 +5,7 @@ import type { UserPhotosGroup } from "../actions/photosActions";
 import type { DatePhotosGroup, IncompleteDatePhotosGroup, PigPhoto } from "../actions/photosActions.types";
 import i18n from "../i18n";
 
-export const EMAIL_REGEX = /^\w+([-.]?\w+\+?\w+[-.]?\w+)?@(\w+-?\w+\.){1,9}[a-z]{2,}$/;
+export const EMAIL_REGEX = /^\w+([-.]?\w+\.?\+?\w+[-.]?\w+)?@(\w+-?\w+\.){1,9}[a-z]{2,}$/;
```
Comment (sickelap) [inline src/util/util.ts]: I might be wrong but I think this makes the email format invalid. I don't think it is valid to have a dot before the plus sign.

I would suggest the following to capture a single character in the middle of the email prefix (including the alias):

`^\w+([-.]?\w+){0,2}(\+?\w+([-.]?\w+){0,2})?@(\w+-?\w+\.){1,9}[a-z]{2,}$`

Comment (AudreyBeard) [inline src/util/util.ts]:
```suggestion
export const EMAIL_REGEX = /^\w+([-.]?\w+){0,2}(\+?\w+([-.]?\w+){0,2})?@(\w+-?\w+\.){1,9}[a-z]{2,}$/;
```

Comment (AudreyBeard) [reply src/util/util.ts]: Oh good catch, thanks! I've implemented this suggestion.

**Comment by sickelap** [comment]
I see sonar does not like the complexity of expression. I would suggest trying to simplify the expression to

`^\w+(?:[.-]?\w+)*@\w+(?:[.-]?\w+)*(?:\.\w{2,})+$`

For that, you would have to remove test cases for emails with aliases (the ones that have a plus sign). Or better, move them to the negative test case examples.

IMO making groups non-capturing is not necessary but I think it's cleaner.

**Comment by derneuere** [comment]
The rule makes sense in general, but for this use case it can be a bit more complex :)

**Pull request merged**